Refactored the code by adding StrykerCli class with run method, Updat…

bicep opened the pull request:
1) Created StrykerCli class with run method
2) Edited binary file by instantiating class
3) Deleted stryker-cli files

simondel commented on packages/stryker/src/StrykerCli.ts:
```diff
@@ -0,0 +1,93 @@
+  private argv: string[];
+
+  constructor(argv: string[]) {
+    this.argv = argv;
```
One of the great things about TypeScript is that you don't need this kind of useless code that you see in many languages. You can change the constructor argument to `private argv: string[]` and it will set the private variable `argv` to the argument so you can remove this line, and the declaration of `private argv: string[];` above the constructor 🎉

simondel commented on packages/stryker/src/StrykerCli.ts:
```diff
@@ -0,0 +1,93 @@
+import * as log4js from 'log4js';
+
+const log = log4js.getLogger('stryker-cli');
+let command: string = '', strykerConfig: string | null = null;
```
We can move these two variables inside of the class so they become properties :)

simondel commented on packages/stryker/bin/stryker:
```diff
@@ -5,4 +5,6 @@
\ No newline at end of file
+//require('../src/stryker-cli');
+const StrykerCli = require('../src/StrykerCli');
+new StrykerCli(process.argv).run()
```
You're missing a semicolon here

simondel commented on packages/stryker/bin/stryker:
```diff
@@ -5,4 +5,6 @@
-require('../src/stryker-cli');
\ No newline at end of file
+//require('../src/stryker-cli');
+const StrykerCli = require('../src/StrykerCli');
```
The constructor function that you export in the StrykerCli file is added to the property `default` so I believe this line should be `const StrykerCli = require('../src/StrykerCli').default;`

bicep commented:
@simondel  Thank you so much for your lengthy reply and guidance. I'll work on them right away and ask questions if I have any!

simondel commented:
Thanks for the update! I noticed that there are still some package.lock.json files, you can see them [here](https://github.com/stryker-mutator/stryker/pull/405/files).
To answer your questions:
1. It turns out it's not in our gitignore... I thought it was.
2. @nicojs and I personally use Visual Studio Code but you are free to use anything you'd like of course. The best way to test the code is to compile it and then run the tests, so `npm run build` and then `npm test`
3. Using `npm run build` if you want to run tslint as well. If you want to compile in the background and not run tslint (which causes the build to fail right now), you could run `npm start`.

bicep commented:
Thanks for your reply! Okay, so I have added the package-lock.json files to gitignore. They should be removed now. When I ran `git rm -r --cached .` and restaged the files with `git add .`, I think I ended up also unstaging files like these:

	deleted:    packages/stryker/testResources/globTestFiles/sample/src/Add.js
	deleted:    packages/stryker/testResources/globTestFiles/sample/src/Array.js
	deleted:    packages/stryker/testResources/globTestFiles/sample/src/Circle.js
	deleted:    packages/stryker/testResources/globTestFiles/sample/src/Error.js

Also am I supposed to do these two things every time before I push a new commit to this branch?
2) fetch+merge local master with the main stryker repo through upstream
3) rebase my branch

I have been doing that per this link (https://gist.github.com/Chaser324/ce0505fbed06b947d962) but this time I was not allowed to rebase. It warned that all the package-lock.json files which are untracked could be replaced.

simondel commented:
Hi! It seems like you accidentally deleted too much. You could remove your last commit using
```
git reset HEAD~1
git push --force
```
Then I would suggest manually deleting the package.lock.json files and adding it to the .gitignore. We squash all pull requests so they won't be in our version history :)

bicep commented:
I'm not sure what's going on with the git commands. Why would I push my code to the repo `git push --force` when I am trying to revert back to the last commit?

bicep commented:
Gah! Something happened as I was fiddling with git and I think I lost all the files..
